tests/test_rewriter.py

```diff
@@ -441,3 +442,34 @@ def transformation(module: ModuleOp, rewriter: Rewriter) -> None:
         rewriter.replace_op(return_op, [new_op])
 
     rewrite_and_compare(prog, expected, transformation)
+
+
+# Test erase operation
+def test_erase_op():
+    prog = """\
+"builtin.module"() ({
+  %0 = "arith.constant"() {"value" = 42 : i32} : () -> i32
+  %1 = "arith.addi"(%0, %0) : (i32, i32) -> i32
+}) : () -> ()
+"""
+
+    expected = """\
+"builtin.module"() ({
+  %0 = "arith.addi"(%1, %1) : (i32, i32) -> i32
+}) : () -> ()
+"""
+
+    def transformation_safe(module: ModuleOp, rewriter: Rewriter) -> None:
+        constant_op = module.ops.first
+        assert constant_op is not None
+        rewriter.erase_op(constant_op, safe_erase=True)
+
+    def transformation_unsafe(module: ModuleOp, rewriter: Rewriter) -> None:
+        constant_op = module.ops.first
+        assert constant_op is not None
+        rewriter.erase_op(constant_op, safe_erase=False)
+
+    with pytest.raises(Exception):
+        rewrite_and_compare(prog, expected, transformation_safe)
+
+    rewrite_and_compare(prog, expected, transformation_unsafe)
```

Comment:
This feels slightly off to me, what are the guarantees about the state of things after raising an error? I don't think we make a guarantee that the module will be in a usable state. It might be worth swapping this line with the two above

Reply:
Reading the source code of ```rewrite_and_compare```, when it is called, the string will be parsed into a newly created module.

Reply:
Anyway, I can swap those lines if you think it is more appropriate.

Reply:
Ah, I missed that they are strings. In that case it's fine, but maybe worth swapping since it's low overhead and might avoid similar confusion in the future. But I don't feel very strongly about this.